src/managementgroups/azext_managementgroups/custom.py

```diff
@@ -0,0 +1,109 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core._profile import Profile
+
+
+def _register_rp(cli_ctx, subscription_id=None):
+    rp = "Microsoft.Management"
+    from azure.cli.core.commands.client_factory import get_mgmt_service_client
+    from azure.cli.core.profiles import ResourceType
+    import time
+    rcf = get_mgmt_service_client(
+        cli_ctx,
+        ResourceType.MGMT_RESOURCE_RESOURCES,
+        subscription_id)
+    rcf.providers.register(rp)
+    while True:
+        time.sleep(10)
+        rp_info = rcf.providers.get(rp)
+        if rp_info.registration_state == 'Registered':
+            break
```

Comment:
RP registration should happen automatically. I don't see anyone else doing this manually.
Is this not the case?
https://github.com/Azure/msrestazure-for-python/blob/master/msrestazure/tools.py#L41

Reply:
@derekbekoe RP registration does not happen for tenant-level resources like ours and subscription-level resources as well. The issue with the auto registration implementation is that an exception is caught to determine whether an RP is registered or not. You can see it [here](https://github.com/Azure/msrestazure-for-python/blob/master/msrestazure/tools.py#L64). However, the "MissingSubscriptionRegistration" exception is not given out for tenant or subscription level resources. Powershell has the same issue with their implementation and they plan to work on it in the near future.

Reply:
@rajshah11 Interesting. What will we get from Azure if your provider is not registered then?
You say that PS plan to work on it, what will be the approach?
Depending of your answer, I might just patch directly msrestazure and release an update, instead of doing this here. since doing it here is good for CLI, but not for SDK users.

Reply:
@lmazuel Yes. Patching msrestazure is better as other services in our team like Reservations and Addons also have the same issue. However, added this temporarily. Most probably ARM just doesn't give any exception for "untracked" resources and then our calls fail because of the "Unregistered" state. I will try to find out what ARM returns and get back to you. Thanks.

Reply:
We agreed to address this in a future release of the extension.